src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -118,10 +144,10 @@ const PBCT  = Union{PBC,  NTuple{<:Any, <:PBC}}
 const MCBCT = Union{MCBC, NTuple{<:Any, <:MCBC}}
 const DCBCT = Union{DCBC, NTuple{<:Any, <:DCBC}}
 
-# Distributed halos have to be filled for last in case of 
-# buffered communication. Hence, we always fill them last
-
-# The reasoning for filling Periodic after Flux, Value, Gradient 
+# Distributed halos have to be filled last because of
+# buffered communication. 
+ 
+# Periodic is handled after Flux, Value, Gradient because
```

Comment:
I thought this was fixed in this PR?

Reply:
That is a much more disruptive change which is nice to have but I believe we should approach it in another dedicated PR (see issue #3342)

Reply:
Alright